### PR TITLE
D19 3 mail editor recipients

### PR DESCRIPTION
**Pull request by tanx**
### NOT YET READY TO MERGE

Opened the PR early for discussion. See below...

### CODE REVIEW

File: common/lib/controller/encrypt.controller.js [hunk 1]
```diff
@@ -30,38 +30,29 @@ define(function(require, exports, module) {
     sub.SubController.call(this, port);
     this.editorControl = null;
     this.recipientsCallback = null;
+    // register event handlers
+    this.on('eframe-recipients', this.displayRecipientProposal);
+    this.on('eframe-display-editor', this.openEditor);
```
Comment (tanx) [inline common/lib/controller/encrypt.controller.js]: @toberndo I wanted to get your feedback on the new event handling style before using it broadly. Events can now be registered like so. I stole the syntax from angular.js. This is much more concise and easier to test than ...

File: common/lib/controller/encrypt.controller.js [hunk 2]
```diff
@@ -30,38 +30,29 @@ define(function(require, exports, module) {
...
-          // TODO show error, fix modalActive on FF
-        } else {
-          this.openEditor(msg.text);
-        }
-        break;
-      default:
-        console.log('unknown event', msg);
-    }
-  };
-
```
Comment (tanx) [inline common/lib/controller/encrypt.controller.js]: ... the old way of doing things in a massive god-event-handler.

File: common/lib/controller/sub.controller.js [hunk 3]
```diff
@@ -63,8 +64,46 @@ define(function(require, exports, module) {
...
+   * @param  {Object} options         Contains message attributes and data
+   */
+  SubController.prototype.handlePortMessage = function(options) {
+    options = options || {};
+    if (this._handlers.has(options.event)) {
+      this._handlers.get(options.event).call(this, options);
+    } else {
+      console.log('Unknown event', options);
+    }
+  };
```
Comment (tanx) [inline common/lib/controller/sub.controller.js]: As documented. Current controller code does not need to be refactored. Controllers can continue to override the abstract function `handlePortMessage`. But if they don't, the new style is assumed.

File: common/lib/controller/encrypt.controller.js (hunk 1 above)
Comment (toberndo) [inline common/lib/controller/encrypt.controller.js]: @tanx Looks good. Great stuff!

**Comment by tanx** [comment]
Closing, creating a new PR for D19-2 + D19-3.

**Pull request closed without merging**